macaroons/service_test.go

```diff
@@ -0,0 +1,134 @@
+package macaroons_test
+
+import (
+	"testing"
+	"path"
+	"os"
+	"context"
+	"io/ioutil"
+	"encoding/hex"
+
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"gopkg.in/macaroon-bakery.v2/bakery/checkers"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+	"google.golang.org/grpc/metadata"
+)
+
+var (
+	testOperation = bakery.Op{
+		Entity: "testEntity",
+		Action: "read",
+	}
+	defaultPw = []byte("hello")
+)
+
+// setupTestRootKeyStorage creates a dummy root key storage by
+// creating a temporary macaroons.db and initializing it with the
+// default password of 'hello'. Only the path to the temporary
+// DB file is returned, because the service will open the file
+// and read the store on its own.
+func setupTestRootKeyStorage(t *testing.T) string {
+	tempDir, err := ioutil.TempDir("", "macaroonstore-")
+	if err != nil {
+		t.Fatalf("Error creating temp dir: %v", err)
+	}
+	db, err := bolt.Open(path.Join(tempDir, "macaroons.db"), 0600,
```

Comment:
Creating a temp file then reading from it is performing IO, so it doesn't seems like this can be considered a unit test but rather than integration test. Unit tests shouldn't fail based upon arbitrary file system conditions.

Note that it isn't really possible to create a Bolt DB without interacting with the file system, so currently macaroons/server can't actually be unit tested.

Reply:
I get your point and I agree that it's not really nice to rely on the file system for a unit test.
But as it seems to be the default way in lnd (have a look at `macaroons/store_test.go`, `routing/pathfind_test.go` or `walletunlocker/service_test.go`) I did it the same way.